app/helpers/form_helper.py

```diff
@@ -128,7 +128,9 @@ def clear_other_text_field(data, questions_for_block):
     """
     form_data = MultiDict(data)
     for question in questions_for_block:
-        for answer in question['answers']:
+        for answer in question.get('answers', []):
+            if answer.get('type') == 'Currency':
+                form_data[answer['id']] = form_data.get(answer['id'], '').replace(',', '')
```

Comment:
This shouldn't be done here, as it has nothing to do with the `other text field` functionality. It should be moved to the `post_form_for_location` or `generate_form` method.